pkg/k8s/statedb.go

```diff
@@ -174,6 +182,14 @@ type k8sReflector[Obj any] struct {
 }
 
 func (r *k8sReflector[Obj]) run(ctx context.Context, health cell.Health) error {
+	if r.CRDSync != nil {
+		// Wait for the CRD to be registered.
+		health.OK("Waiting for CRD registration")
```

Comment:
`CRDSync` here is https://github.com/cilium/cilium/blob/main/pkg/k8s/synced/crd.go#L116, so it's waiting for all of them.

Reply:
Ah, right, makes sense. Could reword to "Waiting for all CRDs to be registered" maybe, since the rest of the `OnDemand` is specific to one resource whereas this is ~global.